new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Nginx.java

```diff
@@ -0,0 +1,159 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import io.kubernetes.client.custom.IntOrString;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressPath;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1HTTPIngressRuleValue;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1Ingress;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressBackend;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressList;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressRule;
+import io.kubernetes.client.openapi.models.ExtensionsV1beta1IngressSpec;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Helm;
+import oracle.weblogic.kubernetes.actions.impl.primitive.HelmParams;
+import oracle.weblogic.kubernetes.actions.impl.primitive.Kubernetes;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+
+/**
+ * Utility class for Nginx ingress controller.
+ */
+public class Nginx {
+
+  private static final String INGRESS_API_VERSION = "extensions/v1beta1";
+  private static final String INGRESS_KIND = "Ingress";
+  private static final String INGRESS_NGINX_CLASS = "nginx";
+
+  /**
+   * Install Nginx Helm chart.
+   *
+   * @param params the parameters to Helm install command such as release name, namespace, repo url or chart dir,
+   *               chart name and chart values
+   * @return true on success, false otherwise
+   */
+  public static boolean install(NginxParams params) {
+    return Helm.install(params.getHelmParams(), params.getValues());
+  }
+
+  /**
+   * Upgrade Nginx Helm release.
+   *
+   * @param params the parameters to Helm upgrade command such as release name, namespace and chart values to override
+   * @return true on success, false otherwise
+   */
+  public static boolean upgrade(NginxParams params) {
+    return Helm.upgrade(params.getHelmParams(), params.getValues());
+  }
+
+  /**
+   * Uninstall Nginx Helm release.
+   *
+   * @param params the parameters to Helm uninstall command such as release name and namespace
+   * @return true on success, false otherwise
+   */
+  public static boolean uninstall(HelmParams params) {
+    return Helm.uninstall(params);
+  }
+
+  /**
+   * Create an ingress for the WebLogic domain with domainUid in the specified domain namespace.
+   * This method assumes the cluster name is "cluster-1" and the managed server port is 8001.
+   *
+   * @param domainNamespace the WebLogic domain namespace in which the ingress will be created
+   * @param domainUid the WebLogic domainUid which is backend to the ingress
+   * @return true on success, false otherwise
+   * @throws ApiException if Kubernetes client API call fails
+   */
+  public static boolean createIngress(String domainNamespace, String domainUid) throws ApiException {
```

Comment:
feels like the cluster name should be a parameter

Reply:
probably the port too

Reply:
Added a cluster name and managed server port as parameters. Removed the createIngress(domainNamespace, domainUid) method.